builtin/debug_test.mbt

```diff
@@ -12,71 +12,152 @@
-// }
+test "debug function with Option[Int]" {
+  let opt = Option::Some(42)
+  let buf = Buffer::new()
+  opt.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|Some(42)
+    ,
+  )!
+  // This should cover the line where the debug function is called with an Option[Int]
+}
 
-// test "debug function with Result[Int, String]" {
-//   let res : Result[Int, String] = Result::Ok(42)
-//   debug(res)
-//   // This should cover the line where the debug function is called with a Result[Int, String]
-// }
+test "debug function with Result[Int, String]" {
+  let res : Result[Int, String] = Result::Ok(42)
+  let buf = Buffer::new()
+  res.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|Ok(42)
+    ,
+  )!
+  // This should cover the line where the debug function is called with a Result[Int, String]
+}
 
-// test "debug function with Ref[Int]" {
-//   let ref_int : Ref[Int] = { val: 42 }
-//   debug(ref_int)
-//   // This should cover the line where the debug function is called with a Ref[Int]
-// }
+test "debug function with Ref[Int]" {
+  let ref_int : Ref[Int] = { val: 42 }
+  let buf = Buffer::new()
+  ref_int.debug_write(buf)
+  inspect(buf, content="{val: 42}")!
+  // This should cover the line where the debug function is called with a Ref[Int]
+}
 
-// test "debug function with FixedArray[Int]" {
-//   let arr = [1, 2, 3, 4, 5]
-//   debug(arr)
-//   // This should cover the line where the debug function is called with a FixedArray[Int]
-// }
+test "debug function with FixedArray[Int]" {
+  let arr = [1, 2, 3, 4, 5]
+  let buf = Buffer::new()
+  arr.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|[1, 2, 3, 4, 5]
+    ,
+  )!
+  // This should cover the line where the debug function is called with a FixedArray[Int]
+}
 
-// test "debug function with Array[Int]" {
-//   let arr = [1, 2, 3, 4, 5]
-//   debug(arr)
-//   // This should cover the line where the debug function is called with an Array[Int]
-// }
+test "debug function with Array[Int]" {
+  let arr = [1, 2, 3, 4, 5]
+  let buf = Buffer::new()
+  arr.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|[1, 2, 3, 4, 5]
+    ,
+  )!
+  // This should cover the line where the debug function is called with an Array[Int]
+}
 
-// test "debug function with Map[Int, String]" {
-//   let map = Map::new()
-//   map.set(1, "one")
-//   map.set(2, "two")
-//   debug(map)
-//   // This should cover the line where the debug function is called with a Map[Int, String]
-// }
+test "debug function with ArrayView[Int]" {
+  let arr = [1, 2, 3, 4, 5]
+  let arrayview = arr[1:2]
+  let buf = Buffer::new()
+  arrayview.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|[2]
+    ,
+  )!
+}
+
+test "debug function with Map[Int, String]" {
+  let map = Map::new()
+  map.set(1, "one")
+  map.set(2, "two")
+  let buf = Buffer::new()
+  map.debug_write(buf)
+  inspect(
+    buf,
+    content=
+      #|(69681622,1,"one"),prev:_,next:-236984087)
```

Comment:
Should we change it to `(key: 69.., index: 1, content:"xx"),...`?

@Yoorkin

Reply:
we can make it align with the map literal: `{ key1: value1, key2: value2 }`